webapp/src/components/admin_settings/teams_input/teams_input.jsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import debounce from 'lodash/debounce';
+import AsyncSelect from 'react-select/async';
+
+// TeamsInput searches and selects teams displayed by display_name.
+// Teams prop can handle the team object or strings directly if the team object is not available.
+// Returns the selected team ids in the `OnChange` value parameter.
+export default class TeamsInput extends React.Component {
```

Comment:
Not in scope with this PR, but it would be interesting to see if we can generalize this input component. Beside `formatOptionLabel` and `getOptionValue`, everything else is identical. It would be great if we would pass those as props and it can be reused for both teams and users. It could also possibly solve https://github.com/mattermost/mattermost-plugin-custom-attributes/issues/28

Reply:
@nathanaelhoun would you be interested in giving https://github.com/mattermost/mattermost-plugin-custom-attributes/issues/28 a shot and possibly exploring the approach above that would apply for users, groups and teams?